### DISCUSSION TITLE
Minio upgrade

### DISCUSSION THREAD

**Discussion by lcperez86** [minio/minio#13931] [Q&A]
Hello
I need to upgrade my minio version, checking the docs, I noticed that there are two ways to do it, replacing the binary with the version you want or using "mc admin update", so keeping this in mind:

1. how do I know if my minio version support "mc admin update", I checked the release notes and there is not something related to it "if there is not comments about should I suppose that "mc admin update " is supported"
2. is there any recommendation or update path to follow for an update, the version that I have is RELEASE.2021-02-14T04-01-33Z

Thanks in advance

**Comment by harshavardhana** [comment]
> * how do I know if my minio version support "mc admin update", I checked the release notes and there is not something related to it "if there is not comments about should I suppose that "mc admin update " is supported"

`mc admin update` only works if MinIO can download binaries from https://dl.minio.io/ and write it locally where the binary is running in-place. 

If you do not have proper permissions in place this can fail and not work. 

If you are in a bare-metal setup you should download the binary from https://dl.minio.io/ and then replace all the binaries `cp -f ./minio.new /usr/bin/minio`  (change this properly for whatever is local path) and then do `mc admin service restart` 

Is this a production cluster @lcperez86 ?

**Reply by lcperez86** [reply]
hello, yeah it is a prod env, so, i want to be sure about the process, i think I need to backup the binary after update it to have a roll back plan.

**Reply by lcperez86** [reply]
I was able to download the package using 
 curl -O https://dl.minio.io/server/minio/release/ from the minio machine that i want to update in that case I suposse that i can use the mc admin update command,

**Reply by harshavardhana** [reply]
> hello, yeah it is a prod env, so, i want to be sure about the process, i think I need to backup the binary after update it to have a roll back plan.

There is no `rollback` allowed @lcperez86 - you should test this first in a staging/UAT environments

NOTE: MinIO is AGPLv3 - if you are using this for commercial purposes you will be under violation of the license terms - please visit https://min.io/pricing for clarification.  

Do check with your lawyers for more detailed clarification.

**Reply by lcperez86** [reply]
Ok, I thought that I could save the binary after the update  as roll back if something goes wrong with the update.
Legal team has everything in order
Thanks

**Reply by harshavardhana** [reply]
> Ok, I thought that I could save the binary after the update as rollback if something goes wrong with the update.

We only allow move forward,  all the newer releases read all the oldest data.  Not the other way around.
